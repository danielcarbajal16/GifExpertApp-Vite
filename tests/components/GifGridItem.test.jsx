import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Pruebas en GifGridItem', () => {
    const title = "FC Barcelona";
    const url = "https://google.com/image.jpg";
    
    test('Debe de hacer match con el snapshot', () => {
        const { container } = render( <GifGridItem title={ title } url={ url } /> )
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el url y el alt indicado', () => {
        render( <GifGridItem title={ title } url={ url } /> );
        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( title );        
    });

    test('Debe de mostrar el titulo de la imagen en el componente', () => {
        render( <GifGridItem  title={ title } url={ url } />);

        expect( screen.getByText( title ) ).toBeTruthy();
    })
})