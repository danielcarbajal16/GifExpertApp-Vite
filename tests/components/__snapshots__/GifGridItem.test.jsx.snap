// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="FC Barcelona"
      src="https://google.com/image.jpg"
    />
    <p>
      FC Barcelona
    </p>
  </div>
</div>
`;
